common/remote/sockfile.cpp

```diff
@@ -2453,7 +2453,7 @@ class CRemoteFileIO : implements IFileIO, public CInterface
         offset_t ret;
         replyBuffer.read(ret);
 
-        if ((ret==(offset_t)-1) || (ret < len))
+        if ((ret==(offset_t)-1) || ((len != ((offset_t)-1)) && (ret < len)))
```

Comment:
ok, so if len of -1 was passed into appendFile() that means append the entire file, right ?